src/API/Middlewares/HttpExceptionsMiddleware.ts

```diff
@@ -1,23 +1,33 @@
-import * as Koa from 'koa'
-
 import { IllegalArgumentException, NotFoundException } from '@po.et/poet-js'
+import { cond, T } from 'ramda'
 
-export const HttpExceptionsMiddleware: Koa.Middleware = async (context: any, next: () => Promise<any>) => {
+export const HttpExceptionsMiddleware = async (context: any, next: () => Promise<any>) => {
   try {
     await next()
   } catch (exception) {
-    if (exception instanceof IllegalArgumentException) {
-      context.body = exception.message
-      context.status = 422
-    } else if (exception instanceof NotFoundException) {
-      context.body = exception.message
-      context.status = 404
-    } else {
-      if (context.logger)
-        context.logger.error({ exception }, 'Error Caught at Middleware - Will Return Internal Server Error')
+    const { body, status } = getCodeError(exception)
+    context.body = body
+    context.status = status
 
-      context.body = 'Internal Server Error'
-      context.status = 503
-    }
+    executeLogger(context, exception, status)
   }
 }
+
+export const executeLogger = (context: any, exception: Error, status: number) => {
```

Comment:
Isn't the `status` parameter a duplication of `context.status` here?

Maybe we can avoid it completely.

Reply:
fixed